src/Symfony/Component/Console/Terminal.php

```diff
@@ -176,10 +176,12 @@ private static function getConsoleMode(): ?array
      */
     private static function getSttyColumns(): ?string
     {
-        return self::readFromProcess('stty -a | grep columns');
+        $output = self::readFromProcess(['stty', '-a']);
+        // join lines to a single one so we can do a single preg_match on it
+        return str_replace("\n", ' ', $output);
```

Comment:
This isn't needed, e.g. https://3v4l.org/2qHAg
(if you wanted to preserve the behavior I guess it would be something like
```suggestion
        preg_match_all('/.+columns.+/', $output, $matches);

        return implode("\n", $matches[0]);
```
but not really needed either)

Reply:
Then why was there a `grep` in the first place? :weary: 

But you're right.

```diff
--- a/src/Symfony/Component/Console/Terminal.php
+++ b/src/Symfony/Component/Console/Terminal.php
@@ -143,11 +143,11 @@ class Terminal
     private static function initDimensionsUsingStty()
     {
         if ($sttyString = self::getSttyColumns()) {
-            if (preg_match('/rows.(\d+);.columns.(\d+);/i', $sttyString, $matches)) {
+            if (preg_match('/rows.(\d+);.columns.(\d+);/is', $sttyString, $matches)) {
                 // extract [w, h] from "rows h; columns w;"
                 self::$width = (int) $matches[2];
                 self::$height = (int) $matches[1];
-            } elseif (preg_match('/;.(\d+).rows;.(\d+).columns/i', $sttyString, $matches)) {
+            } elseif (preg_match('/;.(\d+).rows;.(\d+).columns/is', $sttyString, $matches)) {
                 // extract [w, h] from "; h rows; w columns"
                 self::$width = (int) $matches[2];
                 self::$height = (int) $matches[1];
@@ -176,9 +176,7 @@ class Terminal
      */
     private static function getSttyColumns(): ?string
     {
-        $output = self::readFromProcess(['stty', '-a']);
-        // join lines to a single one so we can do a single preg_match on it
-        return str_replace("\n", ' ', $output);
+        return self::readFromProcess(['stty', '-a']);
     }
 
     private static function readFromProcess(string|array $command): ?string
```
I added an `s` just in case, so a .* in there so a multiline match wouldn't break things.